Added Github Actions

roelofr opened the pull request:
Although Travis CI works fine too, since this is hosted on GitHub and since you're going to start on a V2, I made the guess to add GitHub Actions support to the repo.

Currently test PHP 5.6 - 7.4, with releases from source (`--prefer-source`) and from releases (`--prefer-dist`), since users are more likely to get assets from the latter.

There's an option to add Windows and Mac OS to the matrix, but it'd result in 18 jobs, which seems  a bit too enthusiastic.

rudiedirkx commented:
Interesting. [This is the result?](https://github.com/roelofr/laravel-form-builder/commit/753909eeb2ce2c5521c63db54fbf6bd6e4b0c738/checks?check_suite_id=315112042) It says there's 1 failure, but without any details.

Does it let Github know that 

> All checks have passed
> 1 successful check

etc?

roelofr commented:
That's the "Upload to Scrutinizer" test, which fails since my repo isn't registered in Scrutinizer.

GitHub should change the icon for failed, but non-blocking checks.

rudiedirkx commented:
And "Does it let Github know ...." ? The current CI feedbacks to Github. Does Actions have that? For PRs?

roelofr commented:
Yep, it gets added to the box that now says "All checks have passed".

rudiedirkx commented:
I like it, and it should work in combination with the current CI, but this is really up to @kristijanhusak IMO.

And it should also trigger `on: pull_request`.

roelofr commented:
Good point, I added it.

rudiedirkx commented:
If you remove php 5.6, I'll merge this. Let's give it a shot.

roelofr commented:
I'll work on this tomorrow (CET), I don't have my laptop with me right now.

roelofr commented:
Okay, it's a wall of commits, but I basically just pinned some versions (don't use an `@master` version, but `@v1` or such), removed php 5.6 and added php 8.0 (while faking we're php 7.4, since Laravel Framework specifies a strict version constraint (eh).

rudiedirkx commented:
Looks good. PHP 8.0 oooeeeh. And now we wait: https://github.com/kristijanhusak/laravel-form-builder/actions/runs/38741353

roelofr commented:
Luckily not too long, since they've already passed :ok_hand:

rudiedirkx commented:
Yeah, they're pretty damn fast =) I'm curious to see this at work in a new real PR. This might be a good excuse to finally fix some bugs!
